File: Java/Math/Collatz.java
/**
* @author Edoardo Ferrari
* @version 1.0
*/

import java.util.*;

/* This program will execute the Collatz sequence */

public class Collatz {

	/**
	* This method will decide the Collatz sequence:
	*
	* @return x/2          if the numbder is even
	* @return (3 * x) + 1  if the number is not even
	*/
	
	public static int isEven(int x) {
		if(x % 2 == 0) {
			return x / 2;
		}

		return (3 * x) + 1;
	}

	/* Main method reads from args and prints the Collatz sequence */

	public static void main(String[] args) {
		int x = Integer.parseInt(args[0]);
		System.out.println(x);
		while(x > 1) {
			x = isEven(x);
			System.out.println(x);
		}
	}
}

File: C/Algorithms/Collatz.c
#include <stdio.h>

int collatz(int n) {
	if(n%2 == 0) {
		return n/2;
	}
	
	return (n*3)+1;
}

int main() {
	int n;
	int l = 1;
	printf("Choose a number: ");
	scanf("%d", &n);
	while(n != 1) {
		printf("%d ", n);
		n = collatz(n);
		l++;
	}
	
	printf("1\n%d\n", l);
}
